src/server/auth/errors.ts

```diff
@@ -163,6 +163,13 @@ export class InsufficientScopeError extends OAuthError {
   static errorCode = "insufficient_scope";
 }
 
+/**
+ * Missing authentication error - The request lack required authentication information.
+ */
+export class MissingAuthenticationError extends OAuthError {
+  static errorCode = "missing_authentication";
```

Comment:
note: this error code is non-standard.  there are a few here that aren't official, and this one isn't returned in any response bodies, so this seems okay.

Another option is to not introduce an error class, and just add an if/else in the Invalid token, but this seems like a reasonable tradeoff to be consistent with the rest of that catch block.